hs_core/templates/search/search.html

```diff
@@ -333,55 +530,136 @@ <h4 class="panel-title">
+                });
+            }
+        }
 
-                if (lat_diff < minFloatingNumber && lng_diff < minFloatingNumber) {
-                    item.title = item.title + "<br />" + test.title;
-                    item.info_link += '<br />' + '"<a href="' + test.get_absolute_url + '">' + test.title + '</a>';
-                    existed = true;
-                }
+        var manualZoomMap = function(){
+            var ne_lat = document.getElementById("id_NElat").value;
+            var ne_lng = document.getElementById("id_NElng").value;
+            var sw_lat = document.getElementById("id_SWlat").value;
+            var sw_lng = document.getElementById("id_SWlng").value;
+
+            var ne = new google.maps.LatLng(ne_lat, ne_lng);
+            var sw = new google.maps.LatLng(sw_lat, sw_lng);
+
+            var customized_bounds = new google.maps.LatLngBounds(sw, ne);
+            map.fitBounds(customized_bounds);
+            var $mapDiv = $("#discover-map");
+
+            var mapDim = {
+                height: $mapDiv.height(),
+                width: $mapDiv.width()
+            }
+            var listener = google.maps.event.addListener(map, "idle", function () {
+                map.setCenter(customized_bounds.getCenter());
+                map.setZoom(getBoundsZoomLevel(customized_bounds, mapDim));
+                google.maps.event.removeListener(listener);
             });
-            if (!existed){
-                test.info_link = '"<a href="' + test.get_absolute_url + '">' + test.title + '</a>';
-                modified_boxes_data.push(test);
+        }
+
+        var updateMapView = function() {
+            update_markers = true;
+            var filtered_results = [];
+            clientUpdateMarkers(filtered_results);
+            setMarkers(filtered_results);
+        }
+
+        var updateListView = function(data) {
+            $('#items-discovered').empty();
+            $("#page-options").empty();
+            var requestURL = "/search/";
+            if (window.location.search.length == 0) {
+                requestURL += "?q=";
+            } else {
+                var textSearch = $("#id_q").val();
+                var searchURL = "?q=" + textSearch;
+                requestURL += searchURL;
+                requestURL += buildURLOnCheckboxes();
             }
+
+            $.ajax({
+                type: "GET",
+                url: requestURL,
+                data: data,
+                dataType: 'html',
+                success: function (data) {
+                    var tableDiv = $("#items-discovered", data);
+                    $("#items-discovered").html(tableDiv);
+                    var pageOptionDiv = $("#page-options", data);
+                    $("#page-options").html(pageOptionDiv);
+                    update_markers = false;
+                }
+                , failure: function (data) {
+                    console.error("fail");
+                }
```

Comment:
@xuezhaokun move ```,``` to the upper line for style consistency. It will be better to write detailed error info out to console rather than just output ```fail```